### ISSUE TITLE
Implement other loss functions

### ISSUE THREAD

**Issue by joewallwork** [Cambridge-ICCS/FTorch#594]
MSE and CrossEntropy are implemented in #591 and #593. Based on these implementations, we could potentially generate the other loss functions using an AI agent.
